themes/default/content/docs/esc/other-integrations/docker.md

```diff
@@ -0,0 +1,126 @@
+---
+title_tag: Integrate with Docker | Pulumi ESC
+title: Docker
+h1: "Pulumi ESC: Integrate with Docker"
+meta_desc: This page provides an overview on how to use Pulumi ESC with Docker.
+weight: 2
+menu:
+  pulumiesc:
+    parent: esc-other-integrations
+    identifier: esc-other-integrations-docker
+---
+
+## Overview
+
+Pulumi ESC integrates with [Docker](https://www.docker.com/) to help developers automatically manage configuration and secrets while running `docker` commands.
+
+## Prerequisites
+
+To complete the steps in this tutorial, you will need to install the following prerequisites:
+
+- the [Pulumi ESC CLI](/docs/esc-cli/)
+- the [Docker CLI](https://www.docker.com/) and shell integration
+
+## Manage environment variables for Docker containers
+
+### Create an ESC environment with environment variables
+
+ESC integrates with `docker` by exporting environment variables from an opened environment. Before you can configure `docker`, you'll need to create an environment that exports environment variables. For example, the environment below fetches AWS credentials via OIDC and exports these credentials in environment variables:
+
+```yaml
+values:
+  aws:
+    login:
+      fn::open::aws-login:
+        oidc:
+          duration: 1h
+          roleArn: <your-oidc-iam-role-arn>
+          sessionName: pulumi-environments-session
+  environmentVariables:
+    AWS_ACCESS_KEY_ID: ${aws.login.accessKeyId}
+    AWS_SECRET_ACCESS_KEY: ${aws.login.secretAccessKey}
+    AWS_SESSION_TOKEN: ${aws.login.sessionToken}
+```
+
+For the purposes of this guide, we'll create a simplified environment to demonstrate your options:
+
+```yaml
+values:
+  environmentVariables:
+    ESC_ORG: You are in the ${context.pulumi.organization.login} organization!
+    ESC_HELLO_USER: Hello, ${context.pulumi.user.login}!
+  files:
+    DOCKER_ENVFILE: |
+      ESC_ORG="You are in the ${context.pulumi.organization.login} organization!"
+      ESC_HELLO_USER="Hello, ${context.pulumi.user.login}!"
```

Comment:
I'd recommend using interpolations here to better demonstrate ESC:

```suggestion
values:
  environmentVariables:
    ESC_ORG: You are in the ${context.pulumi.organization.login} organization!
    ESC_HELLO_USER: Hello, ${context.pulumi.user.login}!
  files:
    DOCKER_ENVFILE: |
      ESC_ORG=${environmentVariables.ESC_ORG}
      ESC_HELLO_USER=${environmentVariables.ESC_HELLO_USER}
```

Reply:
Might also be good to build this environment iteratively to better explain what each property contributes. Maybe start with `environmentVariables`, give the first example, then add the `files` as part of the second example?

Reply:
Agreed with pat’s last comment here - and further - might be nice to do that step of adding files later on in the section that specifically uses files.

Reply:
Rewrote the guide with this feedback